File: nightgrass/src/main/java/com/goodfeel/nightgrass/repo/ProductPhotoRepo.java
package com.goodfeel.nightgrass.repo;

import com.goodfeel.nightgrass.data.ProductPhoto;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

public interface ProductPhotoRepo extends ReactiveCrudRepository<ProductPhoto, Long> {
    Flux<ProductPhoto> findAllByProductId(Long productId);
}

File: nightgrass/src/main/kotlin/com/goodfeel/nightgrass/repo/ProductPhotoRepo.kt
package com.goodfeel.nightgrass.repo

import com.goodfeel.nightgrass.data.ProductPhoto
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux

interface ProductPhotoRepo : ReactiveCrudRepository<ProductPhoto, Long> {
    fun findAllByProductId(productId: Long): Flux<ProductPhoto>
}
